plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/S3SecurityMapping.java

```diff
@@ -96,6 +96,25 @@ public S3SecurityMapping(
         this.endpoint = requireNonNull(endpoint, "endpoint is null");
     }
 
+    private static Predicate<URI> prefixPredicate(URI prefix)
```

Comment:
Please revert this change, since it's an unrelated refactoring / code organization and not logically part of this PR.